app/assets/javascripts/admin/addon/models/theme.js

```diff
@@ -174,6 +185,34 @@ class Theme extends RestModel {
     return field ? field.error : "";
   }
 
+  async changeColors() {
+    const colors = [];
+
+    for (const color of this.colorPalette.colors) {
+      const colorPayload = {
+        name: color.name,
+        hex: color.hex,
+        dark_hex: color.dark_hex,
+      };
+
+      colors.push(colorPayload);
+    }
+
+    const paletteData = await ajax(`/admin/themes/${this.id}/change-colors`, {
+      type: "PUT",
+      data: JSON.stringify({ colors }),
+      contentType: "application/json",
+    });
+    this.owned_color_palette = paletteData;
```

Comment:
I'm curious why we need to set this attribute as it doesn't seem to be used anywhere.